R/utils-get_code_dependency.R

```diff
@@ -148,6 +149,22 @@ fix_shifted_comments <- function(calls) {
 #' which is \code{`<-`(y,x)} instead of traditional `y <- x`.
 #' @keywords internal
 #' @noRd
+remove_dt_assign <- function(calls) {
+  checkmate::assert_list(calls)
+  lapply(calls, function(call) {
+    dt_assign <-
+      which(call$token == "LEFT_ASSIGN" & call$text == ":=")
+    if (length(dt_assign) > 0) {
+      call[-dt_assign, ]
+    } else {
+      call
+    }
```

Comment:
You can replace this whole thing with a call to subset, if you were so inclined.

Reply:
@kpagacz like that?

```r
call <- subset(call, !(token == "LEFT_ASSIGN" & text == ":="))
```

Reply:
https://github.com/insightsengineering/teal.code/pull/234/commits/16fea879698c483776e2105c352db043dcf12928

Reply:
According to documentation `subset` should be avoided, I don't think the problematic case applies here, so feel free to keep it

![image](https://github.com/user-attachments/assets/39b5ff07-bcfb-41be-8c09-4c58972dbbb0)

```suggestion
    call[!(token == "LEFT_ASSIGN" & text == ":="), ]
```